optimum/utils/input_generators.py

```diff
@@ -349,8 +349,12 @@ def generate(self, input_name: str, framework: str = "pt"):
                 None,
                 None,
             )
-
-        return super().generate(input_name, framework=framework)
+        else:
+            sequence_length = self.sequence_length
+            self.sequence_length = 1 if input_name == "decoder_input_ids" else self.sequence_length
```

Comment:
I don't think it is. The question here is what does the decoder do if a sequence length > 1 is passed, apparently it's fine to do it, but in practice in `generate()` it's not what happens so that's why I'm wondering if we should force it.

Reply:
`sequence_length` 1 is okay if the `use_past` is true when exporting the decoder. Otherwise, if `use_past` is False then the model still exports, but starts giving incorrect outputs if `sequence_length > 1` for some models. 

I faced this when doing the whisper export in transformers.

Reply:
Interesting. So better be safe and do this check.

Reply:
Yes it is bad, I moved it, will ping you for review